src/cpp/r/session/graphics/RGraphicsUtils.cpp

```diff
@@ -150,29 +154,48 @@ bool validateRequirements(std::string* pMessage)
 
 std::string extraBitmapParams()
 {
-#if defined(_WIN32)
-
-   // no extra params for windows
-
-#elif defined(__APPLE__)
-
-   return ", type = \"quartz\", antialias=\"default\"";
-
-#else
-
-   // if bitmapType is Xlib then force cairo if we can
-   if (r::options::getOption<std::string>("bitmapType") == "Xlib")
-   {
-      if (r::util::hasRequiredVersion("2.14") &&
-          r::util::hasCapability("cairo"))
-      {
-         return ", type = \"cairo\"";
-      }
-   }
-
+   std::vector<std::string> params;
+   
+   std::string backend = r::options::getOption<std::string>(
+            kGraphicsOptionBackend,
+            "default",
+            false);
+   
+#ifndef __APPLE__
+   // silently ignore 'quartz' on non-macOS platforms
+   // (assume this would arise if one copied UI prefs across different machines)
+   if (backend == "quartz")
+      backend = "default";
+#endif
+   
+#ifndef _WIN32
+   // silently ignore 'windows' on non-Windows platforms
+   // (assume this would arise if one copied UI prefs across different machines)
+   if (backend == "windows")
+      backend = "default";
+#endif
+   
```

Comment:
Do we also need to e.g. ignore `cairo` if the pref was set on a machine that had `cairo` but then read on a machine that didn't? (feels like we should just get the set of supported backends here and double check that the pref is included)